src/StatusQ/Components/StatusToastMessage.qml

```diff
@@ -0,0 +1,110 @@
+import QtQuick 2.14
+import QtQuick.Controls 2.12
+import QtQuick.Layouts 1.12
+
+import QtGraphicalEffects 1.13
+import StatusQ.Core 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Core.Theme 0.1
+
+//TODO finish UI
+Control {
+    id: root
+    width: 343
+    height: !!secondaryText ? 68 : 48
+    anchors.right: parent.right
+    anchors.rightMargin: open ? 0 : -width
+    Behavior on anchors.rightMargin { NumberAnimation { duration: 400 }}
+    opacity: open ? 1.0 : 0.0
+    Behavior on opacity { NumberAnimation { duration: 400 }}
+
+    //TODO check on whether to use states
+    //amd remove if not needed
+    property bool open: false
+    property string primaryText: ""
+    property string secondaryText: ""
+
+    function open(title, subTitle) {
+        root.primaryText = title;
+        root.secondaryText = subTitle;
+        root.open = true;
+    }
+
+    signal close()
+
+    background: Rectangle {
+        id: background
+        anchors.fill: parent
+        color: Theme.palette.statusSelect.menuItemBackgroundColor
+        radius: 8
+        border.color: Theme.palette.baseColor2
+        layer.enabled: true
+        layer.effect: DropShadow {
+            verticalOffset: 3
+            radius: 8
+            samples: 15
+            fast: true
+            cached: true
+            color: Theme.palette.dropShadow
+        }
+    }
+
+    contentItem: Item {
+        anchors.left: parent.left
+        anchors.right: parent.right
+        anchors.leftMargin: 16
+        anchors.rightMargin: 16
+        height: parent.height
+        MouseArea {
+            anchors.fill: parent
+            property real initX: 0.0
+            onPressed: {
+                initX = mouseX;
+            }
+            onMouseXChanged: {
+                if (initX < mouseX) {
+                    root.anchors.rightMargin = -mouseX;
+                }
+            }
+            onReleased: {
+                if (mouseX > (root.width/2)) {
+                    root.close();
+                } else {
+                    root.anchors.rightMargin = 0;
+                }
+            }
+        }
+        RowLayout {
+            anchors.fill: parent
+            spacing: 16
+            Rectangle {
+                implicitWidth: 32
+                implicitHeight: 32
+                Layout.alignment: Qt.AlignVCenter
+                radius: (width/2)
+                color: "#4EBC60"
```

Comment:
Not sure what this is about, but if this is some sort of default color for this component, it should live in n`Theme`.